plugin.video.vrt.nu/resources/lib/vrtplayer/tokenresolver.py

```diff
@@ -77,26 +83,26 @@ def _get_new_xvrttoken(self, path, get_roaming_token):
         if not cred.are_filled_in():
             self._kodi_wrapper.open_settings()
             cred.reload()
+
         data = {'loginID': cred.username, 'password': cred.password, 'sessionExpiration': '-1', 'APIKey': self._API_KEY, 'targetEnv': 'jssdk'}
         logon_json = requests.post(self._LOGIN_URL, data).json()
-        token = None
-        if logon_json['errorCode'] == 0:
-            login_token = logon_json['sessionInfo']['login_token']
-            login_cookie = ''.join(('glt_', self._API_KEY, '=', login_token))
-            payload = {'uid': logon_json['UID'], 'uidsig': logon_json['UIDSignature'], 'ts': logon_json['signatureTimestamp'], 'email': cred.username}
-            headers = {'Content-Type': 'application/json', 'Cookie': login_cookie}
-            cookie_jar = requests.post(self._TOKEN_GATEWAY_URL, headers=headers, json=payload).cookies
-
-            xvrttoken = TokenResolver._create_token_dictionary(cookie_jar)
-            token = xvrttoken['X-VRT-Token']
-            if get_roaming_token:
-                xvrttoken = TokenResolver._get_roaming_xvrttoken(login_cookie, xvrttoken)
-                token = xvrttoken['X-VRT-Token'] if xvrttoken is not None else None
-            json.dump(xvrttoken, open(path, 'w'))
-        else:
+        if logon_json['errorCode'] != 0:
```

Comment:
With this implementation when a login fails it will show an error message. And afterwards the method continues without being logged in. Which will most likely result in an error. Am i missing something?

Reply:
You're right, there was a missing `return None` statement.

Reply:
Fixed.

Reply:
I removed that change, because TBH it is not really important at all.